maven-core/src/main/java/org/apache/maven/internal/transformation/ConsumerPomArtifactTransformer.java

```diff
@@ -77,13 +80,34 @@ public void injectTransformedArtifacts(MavenProject project, RepositorySystemSes
                 Files.createDirectories(buildDir);
                 generatedFile = Files.createTempFile(buildDir, CONSUMER_POM_CLASSIFIER, "pom");
```

Comment:
The suffix should be `.pom`, look at the Javadoc of this method.
It is *not* the extension.

Reply:
This was added in [MNG-7622](https://issues.apache.org/jira/browse/MNG-7622) at [this line](https://github.com/apache/maven/pull/907/files#diff-b0e0e02760045bbda4d9839297d968b0218de307f1c4a35a3262d8cb406e0ea3R70). The newly created files currently are extensionless and suffixed `pom` (see screenshot 1). 

If I suffix the newly created files with `.pom` it will be seen as a `POM` extension in my file explorer (screenshot 2).
Would this not be a different ticket to change that behaviour? I'm happy to change it here if not, but I'll have to check if it impacts other places.

[1]
<img width="533" alt="image" src="https://github.com/apache/maven/assets/27761321/9e64036a-ff9e-482f-9122-71412b113aa1">
[2]
<img width="449" alt="image" src="https://github.com/apache/maven/assets/27761321/eeb2b124-a48c-44d0-b81d-5631814a5a25">
<img width="592" alt="image" src="https://github.com/apache/maven/assets/27761321/809f2d4a-a041-4095-8f3e-30881a36dda5">